#1411 Add "Task" to ISSUE_TEMPLATE and include issue type YAML files

Brijeshthummar02 opened the pull request:
Thank you for contributing to Apache StormCrawler.

issue #1411 

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes

- [x] Is there a issue associated with this PR? Is it referenced in the commit message?

- [x] Does your PR title start with `#XXXX` where `XXXX` is the issue number you are trying to resolve?

- [x] Has your PR been rebased against the latest commit within the target branch (typically main)?

- [x] Is your initial contribution a single, squashed commit?

- [ ] Is the code properly formatted with `mvn git-code-format:format-code -Dgcf.globPattern="**/*" -Dskip.format.code=false`?

### For code changes

- [ ] Have you ensured that the full suite of tests is executed via `mvn clean verify`?
- [ ] Have you written or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file, including the main LICENSE file?
- [ ] If applicable, have you updated the NOTICE file, including the main NOTICE file?

### Note

Please ensure that once the PR is submitted, you check GitHub Actions for build issues and submit an update to your PR as soon as possible.

rzo1 commented on .github/ISSUE_TEMPLATE/bug-report.yml:
```diff
@@ -0,0 +1,91 @@
+  attributes:
+    label: Version
+    description: What version of incubator-stormcrawler are you running?
+    options:
```
I think, that it is sufficient to only list the stormcrawler 3.x releases here.

rzo1 commented on .github/ISSUE_TEMPLATE/bug-report.yml:
```diff
@@ -0,0 +1,91 @@
+  id: version
+  attributes:
+    label: Version
+    description: What version of incubator-stormcrawler are you running?
```
What version of "Apache StormCrawler (Incubating) are you using?"

rzo1 commented on .github/ISSUE_TEMPLATE/bug-report.yml:
```diff
@@ -0,0 +1,91 @@
+  attributes:
+    label: How to reproduce
+    placeholder: |
+      + Which incubator-stormcrawler version to use
```
Which version of Apache StormCrawler (Incubating) version to use.

rzo1 commented on .github/ISSUE_TEMPLATE/feature-request.yml:
```diff
@@ -0,0 +1,40 @@
+name: Feature Request
+title: "[FEATURE] "
+description: Suggest an idea for incubator-stormcrawler
```
Apache StormCrawler (Incubating)

tballison reviewed (commented):
Looks good to me with recent changes. Do we need a plain "task" for things like dependency updates that are more complex than dependabot updates...like dealing with breaking api changes?

rzo1 commented:
> Looks good to me with recent changes. Do we need a plain "task" for things like dependency updates that are more complex than dependabot updates...like dealing with breaking api changes?

I am open for it. Might be useful?

Brijeshthummar02 commented:
@tballison That can be useful can you propose how actually we can start with?

mvolikas commented:
@Brijeshthummar02, I think the new .md files should be added to [rat-excludes](https://github.com/apache/incubator-stormcrawler/blob/main/.rat-excludes) for the checks to pass.

mvolikas reviewed (commented):
@Brijeshthummar02, I think the new .md files should be added to [rat-excludes](https://github.com/apache/incubator-stormcrawler/blob/main/.rat-excludes) for the checks to pass.

Brijeshthummar02 commented:
I have updated it can you check now.

rzo1 commented:
FYI:

[INFO] Rat check: Summary over all files. Unapproved: 5, unknown: 5, generated: 0, approved: 7 licenses.
Warning:  Files with unapproved licenses:
  .github/ISSUE_TEMPLATE/epic.yml
  .github/ISSUE_TEMPLATE/improvement.yml
  .github/ISSUE_TEMPLATE/subtask.yml
  .github/ISSUE_TEMPLATE/bug-report.yml
  .github/ISSUE_TEMPLATE/feature-request.yml

Brijeshthummar02 commented:
@rzo1 i got your point, now do i need to do anything?? above are the files i changed as mentioned by you.

rzo1 commented:
You need to update the RAT excludes pattern. Your change didn't work.

Brijeshthummar02 commented:
@rzo1 i have updated, hopefully it should work can you check it now.

mvolikas reviewed (commented):
@Brijeshthummar02
RAT check still fails in my case.
You can test locally with: `mvn -B --no-transfer-progress -Prat -DskipTests verify -Dskip.format.code=false` (from [maven.yml](https://github.com/apache/incubator-stormcrawler/blob/main/.github/workflows/maven.yml))

Brijeshthummar02 commented:
@mvolikas 
![Screenshot 2025-03-05 202708](https://github.com/user-attachments/assets/ba2992eb-38df-4b83-8465-b9aca734992e)
should i do commit?

mvolikas commented:
@Brijeshthummar02 please do. I guess you added the exclusion directly in the pom.
@rzo1 I think the .rat-excludes is not taken into consideration unless we specify: 
```
<excludesFile>.rat-excludes</excludesFile>
```
in the pom.

rzo1 commented:
Just add it to the POM and maybe remove the file from the root. Should be sufficient, I guess.

Brijeshthummar02 commented:
@mvolikas on running 

-  mvn -B --no-transfer-progress -DskipTests verify  build was successful

![Screenshot 2025-03-05 211005](https://github.com/user-attachments/assets/dc7699d0-23e7-4efb-ab02-5327cd34ef32)

mvolikas commented:
> @mvolikas we good with the changes??

Yes. You could also try removing the `.rat-excludes` completely and having all the required exclusions in the `pom`, as @rzo1 suggested. Then you can remove the licence headers in the `yml` files.

Brijeshthummar02 commented:
@mvolikas i have made appropriate changes in pom.xml and removed >.rat-excludes and removed  licence headers in the yml files. Could you please have a look.

mvolikas commented on pom.xml:
```diff
@@ -489,31 +489,31 @@ under the License.
+                                <exclude>**/.classpath</exclude> <!-- Eclipse -->
+                                <exclude>**/.project</exclude> <!-- Eclipse -->
+                                <exclude>**/.idea</exclude> <!-- IntelliJ -->
+                                <exclude>**/.rat-excludes</exclude>
```
@Brijeshthummar02 The rat check is still failing...

You have to add the `<exclude>.github/ISSUE_TEMPLATE/*.yml</exclude>` back.
You can also remove `<exclude>**/.rat-excludes</exclude>` now.

Friendly reminder: I think it's a good practice to run the following locally before pushing:
```
mvn clean -B --no-transfer-progress -Prat -DskipTests verify -Dskip.format.code=false
```

Brijeshthummar02 commented:
@mvolikas i am not getting what is failing my build when running mvn clean -B --no-transfer-progress -Prat -DskipTests verify -Dskip.format.code=false, also during running mvn -B --no-transfer-progress -DskipTests verify   build was successful.

i gone through the changes i made and i don't find what is making it so hard to run the 
```
mvn clean -B --no-transfer-progress -Prat -DskipTests verify -Dskip.format.code=false
```

rzo1 commented:
thx.
